### PR TITLE
Match Silent Potion, Prism Powder, Rainbow Powder, and Deodorizer to match retail accuracy at duration of 10min.

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by hooksta4** [comment]
So how am I messing this up? I created a new branch (silentoils) and then pushed to silentoils branch. Created a PR and it sticks the ones from yesterday in there? I dont get it?

**Comment by Xaver-DaRed** [comment]
You have commits from a different PR here
This happened becouse you, at first, made those commits in base, and THEN you forked from said base.

So basically, and excuse the wording, your base is dirty

**Comment by hooksta4** [comment]
> You have commits from a different PR here This happened becouse you, at first, made those commits in base, and THEN you forked from said base.

How do I fix this? Delete that fork and redo it?

**Comment by Xaver-DaRed** [comment]
ok so...
you have right now some PRs not merged.
I would wait until you get all open PRs merged into our base.
THEN, once that is done, I would pull a fresh copy. start clean

This is the easier way of doing it withought going into push and pulls

**Comment by TeoTwawki** [comment]
IT IS TIME! 
To learn about.. 
[GIT REBASE!  ](https://www.google.com/search?hl=en&q=git%20rebase%20remove%20commits)

O.O

https://git-scm.com/docs/git-rebase

This is why you should never touch your forks base branch, so when you make new branches they start without your other branches commits in them.

**Comment by hooksta4** [comment]
> IT IS TIME! To learn about.. [GIT REBASE! ](https://www.google.com/search?hl=en&q=git%20rebase%20remove%20commits)
> 
> O.O
> 
> https://git-scm.com/docs/git-rebase

Did you just quote Lion King?

**Comment by TeoTwawki** [comment]
> > IT IS TIME! To learn about.. [GIT REBASE! ](https://www.google.com/search?hl=en&q=git%20rebase%20remove%20commits)
> > O.O
> > https://git-scm.com/docs/git-rebase
> 
> Did you just quote Lion King?

Possibly. But I don't think they used git.

:D

**Comment by hooksta4** [comment]
> commits in them.

You guys have been awesome, Thanks so much and sorry for the troubles. short story short, just wait until my branch is even with LSB as I am like 4 commits ahead. 

God damnit, I am going to git good at this... git it?

**Comment by hooksta4** [comment]
> od damnit, I am going to git good at this... git it?

I had an extra space that caused the style check to fail. updated locally, should I push or wait until everything is done?

**Comment by TeoTwawki** [comment]
Leave this for now and after work (I get done somewhere after 5pm eastern) look for me on [discord](https://discord.gg/q4YTHkNVh7) and we'll fix this together

**Comment by hooksta4** [comment]
> Leave this for now and after work (I get done somewhere after 5pm eastern) look for me on [discord](https://discord.gg/q4YTHkNVh7) and we'll fix this together

Good deal, thanks!

**Comment by TeoTwawki** [comment]
> hooksta4 deleted the Silentpots branch 6 minutes ago 

@hooksta4 oopsie?

**Comment by hooksta4** [comment]
> > hooksta4 deleted the Silentpots branch 6 minutes ago
> 
> @hooksta4 oopsie?

why yes indeed....

**Comment by hooksta4** [comment]
FFXIcylopedia has rainbow powder at 10 min. Reflected as such.

**Comment by Xaver-DaRed** [comment]
Clopedia also states that it last from 1 second to 10 minutes. so, a range.

We need retail confirmation of this items. Too much conflicting information.

**Comment by hooksta4** [comment]
For which one or all of them ?

**Comment by Xaver-DaRed** [comment]
Since we are changing them all, all of them.

I find it hard to believe that rainbow powder has only 1 min boon over prism

**Comment by hooksta4** [comment]
Fair enough, I would default back to @ShiyoKozuki , the OP to provide that detail?

**Comment by TeoTwawki** [comment]
> Clopedia also states that it last from 1 second to 10 minutes. so, a range.
> 
> We need retail confirmation of this items. Too much conflicting information.

@Xaver-DaRed SE dropped rangeds. But the advantage of rainbow over prism was always the longer duration When SE bumped up prism they may not have done the same sized increase to rainbow - depends on if they increased the base or the total

**Comment by TeoTwawki** [comment]
retail confirmation all of them including rainbow are 10 min now. old duration on rainbow was 3 min non random. wiki's display of 1 second is for the activation time they just have a template that misleads but when you mouse hover it explains.

so all of these are 600 seconds, but when you receive the icon you'd already lost 1 second and it only displays minutes so shows as 9.

File: scripts/globals/items/pinch_of_rainbow_powder.lua
Comment (TeoTwawki) [inline scripts/globals/items/pinch_of_rainbow_powder.lua]: this check isn't needed, there is no harm in deleting if it didn't exist but the delete isn't needed either as invisible already overwrites itself

**Comment by hooksta4** [comment]
A few cleanup things to address, when applying silent pot when already under status effect sneak - does it give message No_effect in retail behavior? Or nothing?

**Comment by Omnione** [comment]
> A few cleanup things to address, when applying silent pot when already under status effect sneak - does it give message No_effect in retail behavior? Or nothing?

Retail uses "Has no effect".

**Comment by hooksta4** [comment]
> > A few cleanup things to address, when applying silent pot when already under status effect sneak - does it give message No_effect in retail behavior? Or nothing?
> 
> Retail uses "Has no effect".

Thanks

**Comment by almuth150** [comment]
Was there a reason for dropping the sneak/invis multiplier? It doesn't appear to be a setting that has fallen into disuse.

**Comment by hooksta4** [comment]
Yes

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]
Please, add back the xi.settings.SNEAK_INVIS_DURATION_MULTIPLIER to durations. Mostly becouse we still support that setting... At least for now. There's no reson to reduce it's functionality.

Other than that, this seems good.

**Comment by hooksta4** [comment]
> 

Ok, I removed it because its linking to settings SNEAK_INVIS_DURATION_MULTIPLIER, which is 1 by default... not gear (which none of the enhance sneak/invis equip is coded yet) but understand why it would want to be back in there. 

Also tested it out a bunch

File: scripts/globals/items/pinch_of_prism_powder.lua
Comment (Xaver-DaRed) [inline scripts/globals/items/pinch_of_prism_powder.lua]: Aligninment: Insert 4 spaces to this line.
Remove the parenthesis ( ) arround 600

File: scripts/globals/items/pinch_of_prism_powder.lua
Comment (TeoTwawki) [inline scripts/globals/items/pinch_of_prism_powder.lua:21]: indentation in this file got moved incorrectly in 2 places.

File: scripts/globals/items/pinch_of_rainbow_powder.lua
Comment (TeoTwawki) [inline scripts/globals/items/pinch_of_rainbow_powder.lua]: here too

File: scripts/globals/items/pinch_of_prism_powder.lua
Comment (Xaver-DaRed) [inline scripts/globals/items/pinch_of_prism_powder.lua]: Alignment: Remove 4 spaces from this line.

File: scripts/globals/items/pot_of_silent_oil.lua
Comment (TeoTwawki) [inline scripts/globals/items/pot_of_silent_oil.lua]: 1. parenthesis not needed
2. now that its not random, it doesn't even need variablized

Comment (TeoTwawki) [reply scripts/globals/items/pot_of_silent_oil.lua]: same in other places

File: scripts/globals/items/pinch_of_rainbow_powder.lua
Comment (Xaver-DaRed) [inline scripts/globals/items/pinch_of_rainbow_powder.lua]: Alignment: Remove 4 spaces from this line.

File: scripts/globals/items/pinch_of_rainbow_powder.lua
Comment (Xaver-DaRed) [inline scripts/globals/items/pinch_of_rainbow_powder.lua]: Aligninment: Insert 4 spaces to this line.
Remove the parenthesis ( ) arround 600

File: scripts/globals/items/pot_of_silent_oil.lua
Comment (Xaver-DaRed) [inline scripts/globals/items/pot_of_silent_oil.lua]: Remove the parenthesis ( ) arround 600

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]
Some styiling changes needed. TY for bringing back the setting. It is meant for server customization and probably rendered useless with this changes, but still.

EDIT: Seems like Teo and I reviewed at the same time. My apologies.

File: scripts/globals/items/pinch_of_rainbow_powder.lua
Comment (hooksta4) [reply scripts/globals/items/pinch_of_rainbow_powder.lua]: removed entire variable.

File: scripts/globals/items/pinch_of_rainbow_powder.lua
Comment (hooksta4) [reply scripts/globals/items/pinch_of_rainbow_powder.lua]: 21 yea?

Comment (hooksta4) [reply scripts/globals/items/pinch_of_rainbow_powder.lua:21]: `item_object.onItemUse = function(target)
    if target:hasStatusEffect(xi.effect.INVISIBLE) then
        target:delStatusEffect(xi.effect.INVISIBLE)
    end
    target:addStatusEffect(xi.effect.INVISIBLE,1,10, math.floor(600 * xi.settings.SNEAK_INVIS_DURATION_MULTIPLIER))
end`

Comment (Xaver-DaRed) [reply scripts/globals/items/pinch_of_rainbow_powder.lua]: yes

Comment (TeoTwawki) [reply scripts/globals/items/pinch_of_rainbow_powder.lua]: actually while here line breaks between blocks of code are nice:
```lua
if conditions then
    code
end

more code
```

**Review by Xaver-DaRed — APPROVED** [review]

**Review by TeoTwawki — APPROVED** [review]

**Pull request merged**
